### PR TITLE
Clean custom app

### PR DESCRIPTION
**Pull request by scottdixon**
Please review @imurchie

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -242,6 +242,51 @@ class SimulatorXcode6 {
     await simctl.eraseDevice(this.udid);
   }
 
+  async cleanCustomApp (appFile, appBundleId) {
+    log.info(`Cleaning app data files`);
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: The can probably be debugs.

File: lib/simulator-xcode-6.js [hunk 2]
```diff
@@ -242,6 +242,51 @@ class SimulatorXcode6 {
...
+
+    var appDirs = await this.getAppDirs(appFile, appBundleId);
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: `let`

File: lib/simulator-xcode-6.js [hunk 3]
```diff
@@ -242,6 +242,51 @@ class SimulatorXcode6 {
...
+      return;
+    }
+
+    for (let dir of appDirs) {
+      log.info(`Deleting directory: '${dir}'`);
+      await fs.rimraf(dir);
+    }
+
+    if (await this.getPlatformVersion() >= 8) {
+      let relRmPath = 'Library/Preferences/' + appBundleId + '.plist';
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: Use string interpolation.

File: lib/simulator-xcode-6.js [hunk 4]
```diff
@@ -242,6 +242,51 @@ class SimulatorXcode6 {
...
+      let rmPath = path.resolve(this.getRootDir(), relRmPath);
+      log.info("Deleting " + rmPath);
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: Same.

File: test/simulator-e2e-specs.js [hunk 5]
```diff
@@ -62,6 +63,47 @@ function runTests (deviceType) {
       await sim.isFresh().should.eventually.equal(true);
     });
 
+    it('should not find any TestApp data or bundle directories on a fresh simulator', async function(){
```
Comment (imurchie) [inline test/simulator-e2e-specs.js]: `function ()` (i.e., space before parentheses).

File: test/simulator-e2e-specs.js [hunk 6]
```diff
@@ -62,6 +63,47 @@ function runTests (deviceType) {
...
+      await simctl.installApp(udid, getAppPath('TestApp'));
+      await simctl.launch(udid, 'io.appium.TestApp');
+
+      let dirs = await sim.getAppDirs('TestApp', 'io.appium.TestApp');
+      dirs.should.have.length(2);
+      dirs[0].should.contain('/Data/');
+      dirs[1].should.contain('/Bundle/');
+    });
+
+    it('should delete custom app data', async function(){
```
Comment (imurchie) [inline test/simulator-e2e-specs.js]: Same.

File: test/simulator-e2e-specs.js [hunk 7]
```diff
@@ -62,6 +63,47 @@ function runTests (deviceType) {
...
     });
 
+    it('should not find any TestApp data or bundle directories on a fresh simulator', async function(){
+      this.timeout(MED_TIMEOUT);
+      let sim = await getSimulator(udid);
+      let dirs = await sim.getAppDirs('TestApp', 'io.appium.TestApp');
+      dirs.should.have.length(0);
+    });
+
+    it('should find both a data and bundle directory for TestApp', async function(){
```
Comment (imurchie) [inline test/simulator-e2e-specs.js]: Same.

**Comment by scottdixon** [comment]
Thanks, updated!

File: lib/simulator-xcode-6.js [hunk 8]
```diff
@@ -242,6 +242,52 @@ class SimulatorXcode6 {
...
+
+    if (appDirs.length === 0) {
+      log.debug("Couldn't find app directories to delete. Probably it's not " +
+               "installed");
+      return;
+    }
+
+    for (let dir of appDirs) {
+      log.debug(`Deleting directory: '${dir}'`);
+      await fs.rimraf(dir);
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: These deletes can be done in parallel. See how it is done at lines 434-435 below.

File: lib/simulator-xcode-6.js [hunk 9]
```diff
@@ -242,6 +242,52 @@ class SimulatorXcode6 {
...
+    }
+
+    if (await this.getPlatformVersion() >= 8) {
+      let relRmPath = `Library/Preferences/${appBundleId}.plist`;
+      let rmPath = path.resolve(this.getRootDir(), relRmPath);
+      log.debug(`Deleting ${rmPath}`);
+      await fs.rimraf(rmPath);
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: This one could be added to the list.

**Comment by Jonahss** [comment]
What do we use the cleanCustomApp() function for?

**Comment by scottdixon** [comment]
https://github.com/appium/appium-ios-driver/blob/5aa4936e1b7f1334c04853af4bcdeb9664a5515d/lib/driver.js#L421-L425

**Comment by Jonahss** [comment]
Hm, so we clear app data but keep the app so they can keep testing? NEat.

**Pull request merged**